FEXCore/Source/Interface/IR/Passes/RedundantFlagCalculationElimination.cpp

```diff
@@ -17,54 +17,203 @@ desc: This is not used right now, possibly broken
 #include <array>
 #include <memory>
 
+// Flag bit flags
+#define FLAG_N (1 << 0)
+#define FLAG_Z (1 << 1)
+#define FLAG_C (1 << 2)
+#define FLAG_V (1 << 3)
+#define FLAG_A (1 << 4)
+#define FLAG_P (1 << 5)
+
+#define FLAG_NZCV (FLAG_N | FLAG_Z | FLAG_C | FLAG_Z)
+#define FLAG_ALL (FLAG_NZCV | FLAG_A | FLAG_P)
+
 namespace FEXCore::IR {
 
+struct FlagInfo {
+  // If set, all following fields are true, used for a quick exit.
+  bool Trivial;
+
+  // Set of flags read by the instruction.
+  unsigned Read;
+
+  // Set of flags written by the instruction. Happens AFTER the reads.
+  unsigned Write;
+
+  // If true, the instruction can be be eliminated if its flag writes can all be
+  // eliminated.
+  bool CanEliminate;
+
+  // If Replacement != OP_DUMMY, the instruction opcode can be replaced with
+  // Replacement if the flag writes can all be eliminated.
+  IROps Replacement;
+};
+
 class DeadFlagCalculationEliminination final : public FEXCore::IR::Pass {
 public:
   bool Run(IREmitter *IREmit) override;
+
+private:
+  struct FlagInfo Classify(IROp_Header *Node);
 };
 
+struct FlagInfo
+DeadFlagCalculationEliminination::Classify(IROp_Header *IROp)
+{
+  switch (IROp->Op) {
+    case OP_ANDWITHFLAGS:
+      return (struct FlagInfo){
```

Comment:
The tag name can be omitted:

```suggestion
      return {
```

Ditto for the other returns

Reply:
Neat